### PR TITLE
Improve sections handling

### PR DESCRIPTION
**Pull request by dennwc**
* Preserve an order of section while decoding the module
* Add a helper to find a custom section by name
* Simplify code paths that read WASM strings by providing a helper
* Support decoding of "name" custom section (debugging information with function names)

### CODE REVIEW

File: wasm/module.go [hunk 1]
```diff
@@ -68,44 +69,16 @@ type Module struct {
...
-	if m.Elements != nil {
-		sections = append(sections, m.Elements)
-	}
-	if m.Code != nil {
-		sections = append(sections, m.Code)
-	}
-	if m.Data != nil {
-		sections = append(sections, m.Data)
-	}
+	return m.Sections
```
Comment (sbinet) [inline wasm/module.go]: we could perhaps just drop `GetSections()` ?

File: wasm/module.go [hunk 2]
```diff
@@ -68,44 +69,16 @@ type Module struct {
...
+}
+
+func (m *Module) GetCustom(name string) *SectionCustom {
```
Comment (sbinet) [inline wasm/module.go]: `s/GetCustom/Custom/` ?

Comment (dennwc) [reply wasm/module.go]: I was also thinking to rename `Other` to `Custom` and leave `GetCustom` as-is

**Comment by dennwc** [comment]
Done! Also squashed changes into corresponding commits.

**Pull request merged**